### PR TITLE
Add PVC for dev.kubeflow.org

### PR DESCRIPTION
**Pull request by pdmack**
Closes kubeflow/kubeflow#854

Hopefully, the d.k.o default StorageClass is good to go? :-)

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/testing/139)

### CODE REVIEW

**Comment by jlewi** [comment]
Did you actually run recreate_app.sh? Can you do so and check in the new ksonnet app? Can you also run ks apply to actually update dev.kubeflow.org?

Bonus points for fixing .gitignore to check in vendor for the ksonnet app.

**Comment by pdmack** [comment]
@jlewi I don't know what "run ks apply to actually update dev.kubeflow.org" means. Adding the vendor dir and trying to `ks apply` led to lots of errors, starting with the seldon bits.

File: dev-kubeflow-org/ks-app/app.yaml [hunk 1]
```diff
@@ -3,32 +3,32 @@ environments:
   default:
     destination:
       namespace: kubeflow
-      server: https://35.188.73.10
+      server: https://10.19.47.136:6443
```
Comment (jlewi) [inline dev-kubeflow-org/ks-app/app.yaml]: This looks like your minikube cluster. We probably need to point at the dev cluster before running recreate_app.sh

**Comment by jlewi** [comment]
This is great thanks for doing it.

By ks apply I meant run `ks apply` on the generated app to deploy it in dev.kubeflow.org

/lgtm
/approve
/hold 

Feel free to submit as is just cancel the hold.

I put the hold in there just in case you want to try updating dev.kubeflow.org first.

The GKE cluster is in GCP project: kubeflow-dev

**Comment by jlewi** [comment]
Review status: 0 of 63 files reviewed at latest revision, 1 unresolved discussion.

---

*[dev-kubeflow-org/ks-app/components/issue-summarization.jsonnet, line 17 at r2](https://reviewable.io/reviews/kubeflow/testing/139#-LDJKR2oCm00smU8eruu:-LDJKR2p2b7t7InsX5PM:b-bqyc4x) ([raw file](https://github.com/kubeflow/testing/blob/ceab16a58ffee87ab716c4d2c3e4341b7a1863ae/dev-kubeflow-org/ks-app/components/issue-summarization.jsonnet#L17)):*
> ```
> endpoint
> ```
Is this correct? Why is pvcName set to endpoint? Looks like a bug since you added the param pvcName

---


*Comments from [Reviewable](https://reviewable.io/reviews/kubeflow/testing/139)*

**Comment by pdmack** [comment]
Review status: 0 of 63 files reviewed at latest revision, 2 unresolved discussions.

---

*[dev-kubeflow-org/ks-app/components/issue-summarization.jsonnet, line 17 at r2](https://reviewable.io/reviews/kubeflow/testing/139#-LDJKR2oCm00smU8eruu:-LDJYYVk71ZwbNRTbzzy:bl16756) ([raw file](https://github.com/kubeflow/testing/blob/ceab16a58ffee87ab716c4d2c3e4341b7a1863ae/dev-kubeflow-org/ks-app/components/issue-summarization.jsonnet#L17)):*
<details><summary><i>Previously, jlewi (Jeremy Lewi) wrote…</i></summary><blockquote>

> ```
> endpoint
> ```
Is this correct? Why is pvcName set to endpoint? Looks like a bug since you added the param pvcName
</blockquote></details>

typo
Sorry, trying to deal with multiple issues spawned from the original PR

---


*Comments from [Reviewable](https://reviewable.io/reviews/kubeflow/testing/139)*

**Comment by jlewi** [comment]
Looks like @pdmack  was just responding to the comment but didn't push a fix yet.
/issue-summarization.jsonnet

**Comment by pdmack** [comment]
/hold cancel
/lgtm

The PV and PVC are definitely created but I think I'll raise a separate issue regarding the fs mount.
```
# oc describe pv pvc-51019469-6048-11e8-9263-42010a8000db
Name:            pvc-51019469-6048-11e8-9263-42010a8000db
Labels:          failure-domain.beta.kubernetes.io/region=us-central1
                 failure-domain.beta.kubernetes.io/zone=us-central1-a
Annotations:     kubernetes.io/createdby=gce-pd-dynamic-provisioner
                 pv.kubernetes.io/bound-by-controller=yes
                 pv.kubernetes.io/provisioned-by=kubernetes.io/gce-pd
StorageClass:    standard
Status:          Bound
Claim:           kubeflow/claim-accounts-2egoogle-2ecom-3apmackinn-40redhat-2ecom
Reclaim Policy:  Delete
Access Modes:    RWO
Capacity:        10Gi
Message:
Source:
    Type:       GCEPersistentDisk (a Persistent Disk resource in Google Compute Engine)
    PDName:     gke-dev-cluster-22fdbd-pvc-51019469-6048-11e8-9263-42010a8000db
    FSType:     ext4
    Partition:  0
    ReadOnly:   false
Events:         <none>
# oc describe pvc claim-accounts-2egoogle-2ecom-3apmackinn-40redhat-2ecom
Name:          claim-accounts-2egoogle-2ecom-3apmackinn-40redhat-2ecom
Namespace:     kubeflow
StorageClass:  standard
Status:        Bound
Volume:        pvc-51019469-6048-11e8-9263-42010a8000db
Labels:        app=jupyterhub
               heritage=jupyterhub
               hub.jupyter.org/username=accounts_2Egoogle_2Ecom_3Apmackinn_40redhat_2Ecom
Annotations:   pv.kubernetes.io/bind-completed=yes
               pv.kubernetes.io/bound-by-controller=yes
               volume.beta.kubernetes.io/storage-provisioner=kubernetes.io/gce-pd
Finalizers:    []
Capacity:      10Gi
Access Modes:  RWO
Events:
  Type    Reason                 Age   From                         Message
  ----    ------                 ----  ----                         -------
  Normal  ProvisioningSucceeded  4m    persistentvolume-controller  Successfully provisioned volume pvc-51019469-6048-11e8-9263-42010a8000db using kubernetes.io/gce-pd
```

**Comment by pdmack** [comment]
Adding vendor/ means the kubeform_spawner.py needed to be blacklisted for pylint. Will port that over to kubeflow/kubeflow.  

@jlewi with your lgtm, would like to move on.
/cc @texasmichelle

**Comment by jlewi** [comment]
/lgtm
/approve

Thank you so much!

**Pull request merged**